content/en/docs/reference/using-api/api-concepts.md

```diff
@@ -85,7 +85,60 @@ For example:
+A given Kubernetes server will only preserve a historical list of changes for a limited time. Clusters using etcd3 preserve changes in the last 5 minutes by default.  When the requested watch operations fail because the historical version of that resource is not available, clients must handle the case by recognizing the status code `410 Gone`, clearing their local cache, performing a list operation, and starting the watch from the `resourceVersion` returned by that new list operation. Most client libraries offer some form of standard tool for this logic. In Go this is called a `Reflector` and is located in the `k8s.io/client-go/cache` package. The following pseudo-code demonstrates how to implement a reflector:
+
+```
+// Unless the client can tolerate arbitrary old resource versions, which may be served from partioned etcd members or stale caches
+// and can be much older than resource versions the client has observed, the client should provide a sufficiently recent resource version here.
+// If the client is certain can tolerage arbitrary old resource versions, it may use "0" for the initial resource version. A resource version of "0"
+// may be served from cache and can have significant scalability and performance benefits.
+lastObservedResourceVersion := initialResourceVersion
+
+while(should continue watching) {
+  send HTTP LIST request with parameter: resourceVersion=lastObservedResourceVersion
+  switch(HTTP response status code) {
+    case 410: // "Gone"
+      // next relist must be for most recent resource version
+      lastObservedResourceVersion = ""
+    case 200: // "OK"
+      
+      // Set the store to the contents of the list response.
+      // This may either be the 1st list, which determines initial state, or it might
+      // be a "relist", which re-establishes state after a watch is dropped.
+      
+      lastObservedResourceVersion = metadata.resourceVersion in HTTP response
+      send HTTP WATCH request with parameter: resourceVersion=lastObservedResourceVersion
+      switch(HTTP response status code) {
+        case 410: // "Gone"
+          // next relist must be for most recent resource version
+          lastObservedResourceVersion = ""
+        }
+        case 200: // "OK":
+          while(watch HTTP response not closed) {
+            foreach(HTTP response chunked response part) {
+              switch(HTTP response status code) {
+                case 410: // "Gone"
+                  // next relist must be for most recent resource version
+                  lastObservedResourceVersion = ""
+                case 200: 
```

Comment:
same here, I thought this was a watch ADDED/MODIFIED/DELETED event, not an http status